src/expr.rs

```diff
@@ -1008,7 +1008,7 @@ fn rewrite_guard(context: &RewriteContext,
                  // the arm (excludes offset).
                  pattern_width: usize)
                  -> Option<String> {
-    if let &Some(ref guard) = guard {
+    if let Some(ref guard) = *guard {
```

Comment:
I think this is somewhat dubious advice for `if let`, I guess at least it is consistent with match, but I feel the original is clearer